package/MDAnalysis/topology/MOL2Parser.py

```diff
@@ -148,9 +150,34 @@ def parse(self, **kwargs):
         resids = []
         resnames = []
         charges = []
-
+        set_charge = True
+        for line in sections['molecule']:
+            if line.strip().lower() == 'no_charges':
+                set_charge = False
+                break
         for a in atom_lines:
-            aid, name, x, y, z, atom_type, resid, resname, charge = a.split()[:9]
+            columns = a.split()
+            if len(columns) >= 9:
+                columns = columns[:9]
+                aid, name, x, y, z, atom_type, resid, resname, charge = columns
```

Comment:
You can condense this in one line.
```suggestion
                aid, name, x, y, z, atom_type, resid, resname, charge = columns[:9]
```